### PR TITLE
repo-updater: Dev friendly DSN

### PR DESCRIPTION
**Pull request by tsenart**
Context: https://sourcegraph.slack.com/archives/C07KZF47K/p1553628760403700

### CODE REVIEW

File: cmd/repo-updater/repos/db.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 	"database/sql/driver"
 	"net/url"
 	"os"
+	"os/user"
```
Comment (keegancsmith) [inline cmd/repo-updater/repos/db.go]: `os/user` used to imply the need of CGO which can cause issues with alpine/docker stuff. However, I see the godoc for it now mentions there is a pure go implementation it falls back on. So this is likely not a problem anymore (my knowledge on this is likely very outdated)

File: cmd/repo-updater/repos/db.go [hunk 2]
```diff
@@ -33,11 +34,29 @@ type TxBeginner interface {
...
+}
+
+// DefaultDSN returns the default DSN used by repo updater.
+func DefaultDSN() *url.URL {
+	username := "postgres"
+
+	user, err := user.Current()
+	if err == nil {
+		username = user.Username
+	}
```
Comment (keegancsmith) [inline cmd/repo-updater/repos/db.go]: On server and production we specify `PGUSER` right? If so then this should be a safe change to make.

Comment (keegancsmith) [reply cmd/repo-updater/repos/db.go]: Yup we do:
- https://sourcegraph.sgdev.org/search?q=PGUSER+f:repo-updater
- https://sourcegraph.com/search?q=PGUSER+f:server+r:/sourcegraph/sourcegraph

File: cmd/repo-updater/repos/db.go [hunk 3]
```diff
@@ -33,11 +34,29 @@ type TxBeginner interface {
...
+
+	return &url.URL{
+		Scheme:   "postgres",
+		User:     url.User(username),
+		Host:     "127.0.0.1:5432",
+		Path:     "/sourcegraph",
```
Comment (felixfbecker) [inline cmd/repo-updater/repos/db.go]: Doesn't this need to read `PGDATABASE`?

**Pull request merged**